soc/Kconfig

```diff
@@ -36,6 +36,9 @@ config SOC_COMPATIBLE_NRF52X
 config SOC_COMPATIBLE_NRF52832
 	bool
 
+config SOC_ESPRESSIF
```

Comment:
Having this `SOC_ESPRESSIF` config is an option to avoid checking for `SOC_ESP32`, `SOC_ESP32S2` and `SOC_ESP32C3`. As there are both `xtensa` and `riscv` support, I have not found a way to create a common definition for Espressif SoCs. `SOC_FAMILY`  won't work as well due to different platforms. Any other suggestion is welcome.

Reply:
The issue this is trying to address has more to do with the `hal_espressif` than the SoC itself.

I would suggest adding `Kconfig.espressif` under `modules` with `config HAS_ESPRESSIF_HAL` that is selected by the relevant SoCs.

Reply:
Sure, that is a nice approach, thanks for the suggestion.